open_soma() alias handling changes

atolopko-czi opened the pull request:
Resolves #433 

- default to stable release
- notify user when an alias is used
- notify user if stable alias is not defined

New output looks like:

```
import cellxgene_census as cc
c = cc.open_soma()
The "stable" Census version is not yet available. Using "latest" Census version instead.
The "latest" release is currently 2023-04-25. Specify 'census_version="2023-04-25"' in future calls to open_soma() to ensure data consistency.

c = cc.open_soma(census_version="stable")
The "stable" Census version is not yet available. Using "latest" Census version instead.
The "latest" release is currently 2023-04-25. Specify 'census_version="2023-04-25"' in future calls to open_soma() to ensure data consistency.

c = cc.open_soma(census_version="latest")
The "latest" release is currently 2023-04-25. Specify 'census_version="2023-04-25"' in future calls to open_soma() to ensure data consistency.

c = cc.open_soma(census_version="2023-04-25")
# silence is bliss

c = cc.open_soma(census_version="bad")
Traceback (most recent call last):
  File "/Users/atolopko/venvs/cellxgene-census/lib/python3.9/site-packages/IPython/core/interactiveshell.py", line 3508, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-7-16de8ccb18df>", line 1, in <module>
    c = cc.open_soma(census_version="bad")
  File "/Users/atolopko/dev/cxg/cell-census/cellxgene-census/api/python/cellxgene_census/src/cellxgene_census/_open.py", line 129, in open_soma
    raise ValueError(
ValueError: The "bad" Census version is not valid. Use get_census_version_directory() to retrieve available versions.

c = cc.open_soma(census_version=None)
Traceback (most recent call last):
  File "/Users/atolopko/venvs/cellxgene-census/lib/python3.9/site-packages/IPython/core/interactiveshell.py", line 3508, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-8-c137ddd80f7f>", line 1, in <module>
    c = cc.open_soma(census_version=None)
  File "/Users/atolopko/dev/cxg/cell-census/cellxgene-census/api/python/cellxgene_census/src/cellxgene_census/_open.py", line 116, in open_soma
    raise ValueError("Must specify either a census version or an explicit URI.")
ValueError: Must specify either a census version or an explicit URI.
```

pablo-gar commented on api/python/cellxgene_census/src/cellxgene_census/_open.py:
```diff
@@ -50,7 +54,7 @@ def _open_soma(locator: CensusLocator, context: Optional[soma.options.SOMATileDB
 def open_soma(
     *,
-    census_version: Optional[str] = "latest",
+    census_version: Optional[str] = None,
```
shouldn't this be
```suggestion
    census_version: Optional[str] = "stable",
```

ebezzi replied:
Since this is a private method, I think handling the default in the body makes sense.

pablo-gar replied:
Gotcha

atolopko-czi replied:
Actually, I can and will revert this to original behavior. I had a reason for doing this as one point in the coding evolution...

atolopko-czi commented:
@ebezzi @pablo-gar Some minor changes made to user messages, default value location in code, and some extra tests. PTAL at the changes since last review and re-approve.

pablo-gar reviewed (approved):
LGTM
